vault/expiration.go

```diff
@@ -928,12 +928,12 @@ func (m *ExpirationManager) Register(req *logical.Request, resp *logical.Respons
 	}
 
 	// Encode the entry
-	if err := m.persistEntry(&le); err != nil {
+	if err := m.persistEntry(ctx, &le); err != nil {
 		return "", err
 	}
 
 	// Maintain secondary index by token
-	if err := m.createIndexByToken(le.ClientToken, le.LeaseID); err != nil {
+	if err := m.createIndexByToken(ctx, le.ClientToken, le.LeaseID); err != nil {
```

Comment:
If we take a long time to get here, we could end up creating a lease (and a generated value) but fail to associate it with a token(?)